lib/china_regions/helpers/form_helper.rb

```diff
@@ -60,29 +61,44 @@ def region_select(object_name, methods, options = {}, html_options = {})
       private
 
       def get_choices(region_klass, method, preselected_choices, index)
-        choices = (index == 0 ? region_klass.where(nil).collect { |p| [ p.name, p.id ] } : [])
+        return preselected_choices[method] if preselected_choices[method]
+
+        index == 0 ? region_klass.where(nil).collect { |p| [ p.name, p.id ] } : []
+      end
 
-        choices = preselected_choices[:cities] if preselected_choices[:cities] && method == :city
-        choices = preselected_choices[:districts] if preselected_choices[:districts] && method == :district
+      def prioritize_choices(priorities, choices)
+        return choices unless priorities.present?
```

Comment:
When given the choice, default to using an `if` instead of an `unless` - negation is tough to follow. This would work well with `return choices if priorities.empty?` and is much clearer.